tools/tpm2_nvread.c

```diff
@@ -123,6 +123,9 @@ static bool nv_read(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
     if (max_data_size > MAX_NV_BUFFER_SIZE) {
         max_data_size = MAX_NV_BUFFER_SIZE;
     }
+    if (max_data_size == 0) {
+        max_data_size = MAX_NV_BUFFER_SIZE;
```

Comment:
I've heard that picking something along the lines of 512 is a safe bet that it will be supported versus going to the MAX size. So I think we should set this to 512. Make a define for this. `#define DEFAULT_NV_BUFFER_SIZE 512`